packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol

```diff
@@ -275,6 +296,92 @@ contract L1CrossDomainMessenger_Test is CommonTest {
+        uint24 _messageLength,
+        uint32 _minGasLimit,
+        uint32 _gasToUse
+    )
+        external
+    {
+        // TODO(#14609): Update this test to use default.isolate = true once a new stable Foundry
+        // release is available that includes #9904. That will allow us to use this test to check
+        // for changes to the EVM itself that might cause our gas formula to be incorrect.
+
+        // Skip if this is a fork test, won't work.
+        skipIfForkTest("L2CrossDomainMessenger doesn't exist on L1 in forked test");
+
+        // Using smaller uint so the fuzzer doesn't give as many massive values that get bounded.
+        // TODO: Known issue, messages above 34k can actually OOG on the receiving side if the
+        // target uses all available gas. Can be resolved by capping data sizes on the XDM or by
+        // increasing the amount of available relay gas to ~100k. If increasing relay gas, should
+        // have the relay gas only increase when the calldata size is large to avoid disrupting
+        // in-flight L2 -> L1 messages.
+        _messageLength = uint24(bound(_messageLength, 0, 34_000));
+
+        // Need more than 500 since GasBurner requires it.
+        // It's ok to try to use more than minGasLimit since the L2CrossDomainMessenger should
+        // catch the revert and store the message hash in the failedMessages mapping.
+        _gasToUse = uint32(bound(_gasToUse, 500, type(uint32).max));
+
+        // Generate random bytes, more useful than having a _message parameter.
+        bytes memory _message = vm.randomBytes(_messageLength);
+
+        // Compute the base gas.
+        // Base gas should really be computed on the fully encoded message but that would break the
+        // expected API, so we instead just add the encoding overhead to the message length inside
+        // of the baseGas function.
+        uint64 baseGas = l1CrossDomainMessenger.baseGas(_message, _minGasLimit);
+
+        // Deploy a gas burner.
+        address target = address(new GasBurner(_gasToUse));
+
+        // Encode the message.
+        bytes memory encoded = Encoding.encodeCrossDomainMessage(
+            Encoding.encodeVersionedNonce(0, 1), // nonce
+            alice, // Sender doesn't matter
+            target,
+            0, // Value doesn't matter
+            _minGasLimit,
+            _message
+        );
+
+        // Count the number of non-zero bytes in the message.
+        uint256 zeroBytesInCalldata = 0;
+        uint256 nonzeroBytesInCalldata = 0;
+        for (uint256 i = 0; i < encoded.length; i++) {
+            if (encoded[i] != bytes1(0)) {
+                nonzeroBytesInCalldata++;
+            } else {
+                zeroBytesInCalldata++;
+            }
+        }
+
+        // Base gas must always be sufficient to cover the floor cost from EIP-7623.
+        assertGt(baseGas, 21000 + ((zeroBytesInCalldata + nonzeroBytesInCalldata * 4) * 10));
```

Comment:
nit, can this really be `assertGte` for greater than or equal?

Reply:
Yes, I'll probably just leave as-is, doesn't really hurt.